experimental/LieAlgebras/src/LieAlgebra.jl

```diff
@@ -719,31 +718,177 @@ function _root_system_and_chevalley_basis(
+      (F.(_matrix(h)) for h in scaled_cartan_elems)...,
+    ]; init=zero_matrix(F, 0, dim(L_F)))
+  # The structure constants of `L_F` w.r.t. the Chevalley basis lie in `R`,
+  # so we can construct the Lie algebra `K` over `R` with the these structure constants.
+  # The basis elements of `K` form a Chevalley basis.
+  K = lie_algebra(
+    R, map(e -> change_base_ring(R, e), _structure_constant_table(L_F, L_F_chev_basis_mat))
+  )
+
+  # Construct an isomorphism f from K to L.
+  # To save computation time, we only compute f on L_alpha for alpha a root.
+  domain_basis_xs = [basis(K, i) for i in 1:n_simple_roots(rs)]
+  domain_basis_ys = [basis(K, n_positive_roots(rs) + i) for i in 1:n_simple_roots(rs)]
+  codomain_basis_xs = copy(canonical_gens_xs)
+  codomain_basis_ys = copy(canonical_gens_ys)
+  for (i, alpha) in enumerate(positive_roots(rs))
+    i <= n_simple_roots(rs) && continue # already set above
+    j, k = 0, 0
+    found = false
+    for outer j in 1:rank(rs)
+      (fl, k) = is_positive_root_with_index(alpha - simple_root(rs, j))
+      if fl
+        found = true
+        break
+      end
+    end
+    @assert found
+    push!(domain_basis_xs, domain_basis_xs[j] * domain_basis_xs[k])
+    push!(domain_basis_ys, domain_basis_ys[j] * domain_basis_ys[k])
+    push!(codomain_basis_xs, codomain_basis_xs[j] * codomain_basis_xs[k])
+    push!(codomain_basis_ys, codomain_basis_ys[j] * codomain_basis_ys[k])
+  end
+
+  # The image of the Chevalley basis of K under f is a Chevalley basis of L.
+  xs = [
+    only(solve(_matrix(domain_basis_xs[i]), _matrix(basis(K, i)); side=:left)) *
+    codomain_basis_xs[i] for i in 1:n_positive_roots(rs)
+  ]
+  ys = [
+    only(
+      solve(
+        _matrix(domain_basis_ys[i]), _matrix(basis(K, n_positive_roots(rs) + i)); side=:left
+      ),
+    ) * codomain_basis_ys[i] for i in 1:n_positive_roots(rs)
+  ]
+
+  return rs, (xs, ys, scaled_cartan_elems)
+end
+
+# TODO: clean the following up, once we have a unified interface for field extensions
+function _field_ext_with_sqrts(F::QQField, elems::Vector{QQFieldElem})
+  i = findfirst(!is_square, elems)
+  if isnothing(i)
+    return F, sqrt.(elems)
+  end
+  Fx, x = polynomial_ring(F; cached=false)
+  F, _ = number_field(x^2 - elems[i]; cached=false)
+  return _field_ext_with_sqrts(F, [F(c) for c in elems])
+end
+
+function _field_ext_with_sqrts(F::NumField, elems::Vector{<:NumFieldElem})
+  @req all(e -> parent(e) === F, elems) "Incompatible number fields"
+  elems_F = elems
+  i = findfirst(!is_square, elems_F)
+  while !isnothing(i)
+    Fx, x = polynomial_ring(F; cached=false)
+    F, _ = number_field(x^2 - elems_F[i]; cached=false)
+    elems_F = [F(c) for c in elems_F]
+    i = findfirst(!is_square, elems_F)
+  end
+  return F, sqrt.(elems_F)
+end
 
-  return R, (xs, ys, hs)
+function _field_ext_with_sqrts(F::FqField, elems::Vector{<:FqFieldElem})
+  @req all(e -> parent(e) === F, elems) "Incompatible number fields"
```

Comment:
```suggestion
  @req all(e -> parent(e) === F, elems) "Incompatible parent fields"
```
This is for finite fields after all.

But isn't the code identical to that for `NumFieldElem`? Perhaps merge the two methods into one then? (for `F::Field, elems::Vector{<:FieldElem}`)

Reply:
Unfortunately not. The function doing the field extension itself is `number_field` or `finite_field`.
@thofma agreed with me on slack that these should eventually get a unified interface, but not right now and by me